### PR TITLE
Add a default method to override credentialsProvider

### LINKED ISSUE #4604: S3Presigner.Builder returns SdkPresigner
### Describe the bug

Take this Scala 2.13 code:

```
 private def getAwsS3PreSigner(config: Config): S3Presigner = {
    val awsAccessKeyId = config.getString("aws.iam.accessKey.id")
    val awsSecretKey   = config.getString("aws.iam.accessKey.secret")
    val regionName     = config.getString("aws.region")

    val awsCredentials: AwsBasicCredentials = AwsBasicCredentials.create(awsAccessKeyId, awsSecretKey)
    val region                              = Region.of(regionName)

    S3Presigner
      .builder()
      .region(region)
      .credentialsProvider(StaticCredentialsProvider.create(awsCredentials))
      .build()
  }
```

With AWS SDK `2.20.162`, this compiles properly. With AWS SDK `2.21.1` this doesn't compile, causing a typing error:
```
[error] [..]/Main.scala:431:13: type mismatch;
[error]  found   : software.amazon.awssdk.awscore.presigner.SdkPresigner
[error]  required: software.amazon.awssdk.services.s3.presigner.S3Presigner
[error]       .build()
[error]             ^
```

### Expected Behavior

I expect that `2.21.1` is backwards compatible with `2.20.162`  and that the builder returns the proper implementation. This is likely caused by  `.region()` and/or `.credentialsProvider()` because without those two statements, it does still compile properly. 

### Current Behavior

Compile-error because the wrong type is returned by `.build()`. Do mind, that other SdkBuilder instances like SNS or CloudWatch all compile as expected.

### Reproduction Steps

Feel free to take the code from above. My apologies for giving you this example in Scala 2.13

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### AWS Java SDK version used

2.21.1

### JDK version used

OpenJDK Runtime Environment Corretto-21.0.0.35.1 (build 21+35-LTS)

### Operating System and version

Fedora Linux 38

### PR DESCRIPTION
**Pull request by sugmanue**
## Motivation and Context

A bug reported by a customer #4604 noticed that the Scala compiler is getting confused about an overridden method with a more specific type. In order to work around that this change defines a default method instead which makes the Scala compiler happy.

Fixes #4604.

## Modifications

Added a default method for the `SdkPresigner.Builder` class.

## Testing

Tested manually using an internal Scala based project.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: .changes/next-release/bugfix-AWSSDKforJavav2-bd91b46.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "category": "AWS SDK for Java v2",
```
[the remark this answers is not in the document]
Comment (gosar) [reply .changes/next-release/bugfix-AWSSDKforJavav2-bd91b46.json:2]: though I'm not sure how that is handled when multiple services are being changed? one change log per service?

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/presigner/S3Presigner.java [hunk 2]
```diff
@@ -586,7 +586,9 @@ interface Builder extends SdkPresigner.Builder {
         Builder region(Region region);
 
         @Override
-        Builder credentialsProvider(AwsCredentialsProvider credentialsProvider);
+        default Builder credentialsProvider(AwsCredentialsProvider credentialsProvider) {
```
[the remark this answers is not in the document]
Comment (sugmanue) [reply services/s3/src/main/java/software/amazon/awssdk/services/s3/presigner/S3Presigner.java:589]: Good catch, I have added `PollyPresigner` as well.

File: .changes/next-release/bugfix-AWSSDKforJavav2-bd91b46.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sugmanue) [reply .changes/next-release/bugfix-AWSSDKforJavav2-bd91b46.json:2]: Not sure what to do about this one, I will let the Java team to chime in.

[the remark this answers is not in the document]
Comment (dagnir) [reply .changes/next-release/bugfix-AWSSDKforJavav2-bd91b46.json:2]: Yep, in this case please create entries for each

[the remark this answers is not in the document]
Comment (sugmanue) [reply .changes/next-release/bugfix-AWSSDKforJavav2-bd91b46.json]: Sounds good, let me cook that up.

**Review by dagnir — APPROVED** [review]

**Pull request merged**